### PR TITLE
Issue #184: Remove dependency on base_local_planner

### PR DESCRIPTION
**Pull request by corot**
Copied footprint_helper from base_local_planner to mbf_costmap_nav.

Code is essentially the same, except for making the methods static and remove Eigen, to simplify calling interface. I kept the original license (BSD) and Copyright (c) (2008, Willow Garage, Inc.). I suppose this is fine, isn't it?

### CODE REVIEW

**Comment by spuetz** [comment]
Nice! Thank you :+1:  One little thing to discuss: With the PR, the mbf_utils depend on costmap_2d, right? Should be fine, or?

**Comment by corot** [comment]
> Nice! Thank you  One little thing to discuss: With the PR, the mbf_utils depend on costmap_2d, right? Should be fine, or?

mmm.... mmm....  I'm reconsidering.... cause mbf_abstract_nav depends on mbf_utils, so it will inherit a dependency on costmap_2d....

I can move the code to mbf_costmap_nav instead. I didn't cause it looks very "patchy", but maybe is better than spreading costmap_2d dependency. I let u decide.

**Comment by spuetz** [comment]
Hmmm...the `mbf_abstract_*` should not depend on `costmap_2d`.

**Comment by spuetz** [comment]
So yes, please move the code to `mbf_costmap_nav`. :) Thank you @corot

**Comment by corot** [comment]
stimt

File: mbf_costmap_nav/CMakeLists.txt [hunk 1]
```diff
@@ -3,18 +3,18 @@ project(mbf_costmap_nav)
...
 find_package(catkin REQUIRED
   COMPONENTS
-  base_local_planner
   dynamic_reconfigure
   geometry_msgs
   mbf_abstract_nav
   mbf_costmap_core
-  mbf_msgs
   mbf_utility
+  mbf_msgs
```
Comment (spuetz) [inline mbf_costmap_nav/CMakeLists.txt]: ?

File: mbf_costmap_nav/CMakeLists.txt [hunk 2]
```diff
@@ -3,18 +3,18 @@ project(mbf_costmap_nav)
...
   nav_core
   nav_msgs
-  roscpp
```
Comment (spuetz) [inline mbf_costmap_nav/CMakeLists.txt]: ?

File: mbf_costmap_nav/CMakeLists.txt [hunk 3]
```diff
@@ -35,19 +35,19 @@ catkin_package(
...
   actionlib
   actionlib_msgs
-  base_local_planner
   dynamic_reconfigure
   geometry_msgs
   mbf_abstract_nav
   mbf_costmap_core
-  mbf_msgs
   mbf_utility
+  mbf_msgs
```
Comment (spuetz) [inline mbf_costmap_nav/CMakeLists.txt]: ?

File: mbf_costmap_nav/CMakeLists.txt [hunk 4]
```diff
@@ -35,19 +35,19 @@ catkin_package(
...
   nav_core
   nav_msgs
   pluginlib
-  roscpp
   std_msgs
   std_srvs
+  costmap_2d
+  roscpp
```
Comment (spuetz) [inline mbf_costmap_nav/CMakeLists.txt]: ?

File: mbf_utility/package.xml [hunk 5]
```diff
@@ -11,18 +11,11 @@
...
-  <build_depend>tf</build_depend>
-  <build_export_depend>geometry_msgs</build_export_depend>
-  <build_export_depend>roscpp</build_export_depend>
-  <build_export_depend>tf</build_export_depend>
-  <exec_depend>geometry_msgs</exec_depend>
-  <exec_depend>roscpp</exec_depend>
-  <exec_depend>tf</exec_depend>
 
+  <depend>roscpp</depend>
+  <depend>geometry_msgs</depend>
```
Comment (spuetz) [inline mbf_utility/package.xml]: Sort alphabetically

**Review by spuetz — CHANGES_REQUESTED** [review]

File: mbf_costmap_nav/CMakeLists.txt (hunk 1 above)
Comment (corot) [reply mbf_costmap_nav/CMakeLists.txt]: Done

File: mbf_costmap_nav/CMakeLists.txt [hunk 6]
```diff
@@ -3,13 +3,13 @@ project(mbf_costmap_nav)
...
   COMPONENTS
-  base_local_planner
+  costmap_2d
   dynamic_reconfigure
   geometry_msgs
   mbf_abstract_nav
   mbf_costmap_core
-  mbf_msgs
   mbf_utility
+  mbf_msgs
```
Comment (spuetz) [inline mbf_costmap_nav/CMakeLists.txt]: m < u

**Review by spuetz — CHANGES_REQUESTED** [review]

**Pull request merged**